ops/model.py

```diff
@@ -62,15 +61,17 @@ def get_unit(self, unit_name):
     def get_app(self, app_name):
         return self._cache.get(Application, app_name)
 
-    @property
-    def bindings(self):
-        if self._bindings is None:
-            binding_keys = list(self._meta.relations) + list(self._meta.extra_bindings)
-            for endpoint_relations in self.relations.values():
-                if endpoint_relations:
-                    binding_keys.extend(endpoint_relations)
-            self._bindings = BindingMapping(binding_keys, self._backend)
-        return self._bindings
+    def get_binding(self, binding_name):
+        """Get a binding without specifying a relation id."""
+        return self.bindings._get_unique(binding_name)
+
+    def get_relation_binding(self, relation):
+        """Get a binding for a specific relation.
+
+        Using this method is preferred if a Relation instance can be provided because network info
+        exposed in a Binding instance can be different depending on whether a relation is a cross-model relation or not.
+        """
+        return self.bindings._get_unique(relation.name, relation.id)
```

Comment:
What's the rationale behind these methods?  How an someone specify a different space for a relation depending on its ID?  If that's impossible, why would we have any other interfaces besides the one that looks up a binding via its relation name?

Reply:
Resolving per our out-of-band discussions.

Context: https://discourse.jujucharms.com/t/multiple-space-bindings-per-endpoint/1999

Per our discussion, I left only the `get_binding` method that will now accept both a string and a Relation:
https://github.com/canonical/operator/pull/109/commits/ada20808fd8fef9d1714e31d08c47043dab0a6c5